src/Ocelot/LoadBalancer/LoadBalancers/RoundRobin.cs

```diff
@@ -30,6 +30,17 @@ public async Task<Response<ServiceHostAndPort>> Lease(HttpContext httpContext)
                     }
 
                     var next = services[_last++];
+
+                    if (next == null)
+                    {
+                        return new ErrorResponse<ServiceHostAndPort>(new ServiceIsInvalidError($"The selected service was null in {nameof(RoundRobin)} during {nameof(Lease)} operation."));
```

Comment:
Very well... But adding of a new error-class is redundant!
Pay your attention that especially for this null-case we have already existed the [ServicesAreNullError](https://github.com/search?q=repo%3AThreeMammals%2FOcelot%20ServicesAreNullError&type=code) class.
Example of usage is here in the `LeastConnection` class ->
https://github.com/ThreeMammals/Ocelot/blob/8c0180a6461984ebd11ceca5544ca9d0e1564975/src/Ocelot/LoadBalancer/LoadBalancers/LeastConnection.cs#L27
so I suggest the following change:
```suggestion
                        return new ErrorResponse<ServiceHostAndPort>(new ServicesAreNullError($"The selected service was null in {nameof(RoundRobin)} during {nameof(Lease)} operation."));
```
Also it will be informative to specify total count and current failed index as an additional information.

Reply:
Fixed in https://github.com/ThreeMammals/Ocelot/pull/2111/commits/bbed7d1aa195f425939cc5f3a69e245f6bdd565e